pkg/serverless/trace/inferredspan/span_enrichment.go

```diff
@@ -232,6 +348,13 @@ func (inferredSpan *InferredSpan) EnrichInferredSpanWithSQSEvent(eventPayload ev
 	inferredSpan.Span.Service = serviceName
 	inferredSpan.Span.Start = startTime
 	inferredSpan.Span.Resource = parsedQueueName
+	if err == nil {
```

Comment:
Would `err == nil` be true if `convertToUint64(parentID)` failed but `convertToFloat64(samplingPriority)` worked?

Reply:
I wonder if we can just combine the code from line 319-340 with these 3 assignments and move them together down to the last step of this method. This way we can just return if an err occurs.